Added Speaker Details Activity

drigio opened the pull request:
Added SpeakerDetailsActivity 
Issue #11 

- [x] Circular Profile Picture
- [x] Name
- [x] Designation
- [x] Introduction
- [x] Social Links

Please review and let me know any changes.


<img src="https://user-images.githubusercontent.com/22986282/62647102-72cb2300-b96d-11e9-93f5-9f0eb6c4e9d4.jpeg" alt="Screenshot" width=250 >

drigio committed:
Merge remote-tracking branch 'PatilShreyas/master'

PatilShreyas commented on app/src/main/java/com/gdg/pune/devfest19/ui/SpeakerDetailsActivity.kt:
```diff
@@ -1,16 +1,126 @@
         setContentView(R.layout.activity_speaker_details)
+        val extras = intent.extras
+        if (extras != null) {
+            speaker.name = extras.getString("speakerName")
```
You are passing every data from intent. Just pass document id from intent and refer it here in `DocumentReference`. Refer code style of `SessionDetailsActivity`. It will be more easy.

drigio replied:
Wouldn't it result in re retrieving the query from firebase?

PatilShreyas replied:
Actually it's good practice in Firebase. If you are using `SnapshotListener` it loads recently loaded cached data.

drigio replied:
Okay great then, will do it!

PatilShreyas commented on app/src/main/java/com/gdg/pune/devfest19/ui/fragment/SpeakersFragment.kt:
```diff
@@ -27,7 +27,7 @@ class SpeakersFragment : AbstractRecyclerFragment() {
 
             override fun onBindViewHolder(viewHolder: SpeakerViewHolder, position: Int, model: Speaker) {
-                viewHolder.bind(model, null)
+                viewHolder.bind(model)
```
Actually, When we are passing document id it should be pass from intent. That intent should be in adapter class not im ViewHolder class. We just have to set listener in ViewHolder class. Refer code of `ScheduleFragment`.

PatilShreyas commented on app/src/main/java/com/gdg/pune/devfest19/viewholder/SpeakerViewHolder.kt:
```diff
@@ -29,6 +31,22 @@ class SpeakerViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
             profileView.setImageResource(R.drawable.ic_account_circle)
         }
+
+        val onClickListener = View.OnClickListener {
```
As I said earlier, intent should be instantiated in adapter class for document id purpose.

PatilShreyas commented:
@drigio PR Looks good. Just few comments.

drigio commented:
Thanks, will start working on it. I'm still new to Firebase design principles.

PatilShreyas commented:
Sure! No problem. You will surely learn new things by this. It'll be helpful for you.

PatilShreyas reviewed (approved):
@drigio Now it looks great! Thanks for this PR. I'll merge it.

PatilShreyas reviewed (changes requested):
Content is not scrolling. We should fix it. (Not urgent, we'll fix it later. Just written here to remember)

drigio committed:
Added SpeakerDetailsActivity

Created the basic layout and UI for the SpeakerDetailsActivity. Added Roboto fonts.
Added SpeakerDetailsActivity

Completed UI.
Added Required Details.
Added Social Links
Added debug build apk


Reformatted code according to Firebase design


Updated debug-apk file


Fixed the ScrollView of introduction

Fixed content overflowing of Speaker introduction

drigio committed:
Updated app-debug apk

drigio commented:
> Content is not scrolling. We should fix it. (Not urgent, we'll fix it later. Just written here to remember)

Fixed It.

PatilShreyas commented:
Thanks @drigio for this PR.
